src/test/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProviderTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.gcs;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemConfiguration;
+import com.google.cloud.hadoop.util.AccessTokenProvider;
+import com.google.cloud.hadoop.util.CredentialFactory;
+import com.google.cloud.hadoop.util.CredentialFromAccessTokenProviderClassFactory;
+import com.google.cloud.hadoop.util.HadoopCredentialConfiguration;
+import com.google.common.collect.ImmutableList;
+import io.cdap.plugin.gcp.common.GCPUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.junit.Assert;
+import org.junit.Test;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Unit Tests for {@link ServiceAccountAccessTokenProvider}.
+ */
+public class ServiceAccountAccessTokenProviderTest {
+
+  @Test
+  public void testServiceAccountAccessTokenProviderIsUsed() throws IOException {
+
+    Map<String, String> authProperties = GCPUtils.generateGCSAuthProperties(null,
+        "filePath");
+    Configuration conf = new Configuration();
+    for (Map.Entry<String, String> prop : authProperties.entrySet()) {
+      conf.set(prop.getKey(), prop.getValue());
+    }
+
+    AccessTokenProvider accessTokenProvider =
+        HadoopCredentialConfiguration.getAccessTokenProvider(conf, ImmutableList.of(
+        GoogleHadoopFileSystemConfiguration.GCS_CONFIG_PREFIX));
+
+    Assert.assertTrue(accessTokenProvider instanceof ServiceAccountAccessTokenProvider);
```

Comment:
nit: `assertThat(accessTokenProvider, instanceOf(ServiceAccountAccessTokenProvider.class));` I would give proper message when failed instead of just "assertion failed"

Reply:
done.